tests/Attribute/Parameter/ToDateTimeImmutableTest.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Hydrator\Tests\Attribute\Parameter;
+
+use DateTime;
+use DateTimeImmutable;
+use DateTimeZone;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\TestCase;
+use stdClass;
+use Yiisoft\Hydrator\ArrayData;
+use Yiisoft\Hydrator\Attribute\Parameter\ToDateTimeImmutable;
+use Yiisoft\Hydrator\Attribute\Parameter\ToDateTimeImmutableResolver;
+use Yiisoft\Hydrator\AttributeHandling\Exception\UnexpectedAttributeException;
+use Yiisoft\Hydrator\AttributeHandling\ParameterAttributeResolveContext;
+use Yiisoft\Hydrator\AttributeHandling\ResolverFactory\ContainerAttributeResolverFactory;
+use Yiisoft\Hydrator\Hydrator;
+use Yiisoft\Hydrator\Result;
+use Yiisoft\Hydrator\Tests\Support\Attribute\Counter;
+use Yiisoft\Hydrator\Tests\Support\Attribute\CounterResolver;
+use Yiisoft\Hydrator\Tests\Support\Classes\CounterClass;
+use Yiisoft\Hydrator\Tests\Support\TestHelper;
+use Yiisoft\Test\Support\Container\SimpleContainer;
+
+final class ToDateTimeImmutableTest extends TestCase
+{
+    public static function dataBase(): array
+    {
+        return [
```

Comment:
Please use `yield` construction to emit test cases

Reply:
done